test/functional/simulator-e2e-specs.js

```diff
@@ -349,6 +349,17 @@ describe('advanced features', function () {
       });
     });
   });
+
+  describe('Permission', function () {
+    it('should set and get', async function () {
+      await sim.setPermission('com.apple.Maps', 'microphone', 'yes');
+      await sim.getPermission('com.apple.Maps', 'microphone').should.eventually.eql('yes');
+      await sim.setPermission('com.apple.Maps', 'microphone', 'no');
+      await sim.getPermission('com.apple.Maps', 'microphone').should.eventually.eql('no');
+      await sim.setPermission('com.apple.Maps', 'microphone', 'unset');
+      await sim.getPermission('com.apple.Maps', 'microphone').should.eventually.eql('unset');
+    });
```

Comment:
Below combination worked:
Functional Tests / test (13.1, iPhone X, 15)
Functional Tests / test (14.2, iPhone 14, 16.2)


Locally the same one worked with Xcode 15 and iOS 17.0, and Xcode 14.3 and iOS 16.4 simulators. So... in current our supported Xcode and iOS versions probably good with this...?

Reply:
In the actual issue there was an issue with contacts permission. Have you checked that one as well?

Reply:
Oh, contacts. Default apps do not require `contacts` permission. Let me create a thin app

Reply:
You're correct, it seems like contacts did not work with code 15 x ios 17 as well. It looks like we should use wix only for contacts right now

Reply:
Hm, but `applesimutils` also did not work for contacts

Reply:
So, probably we could keep wix's one for below services

```
    camera=YES|NO|unset
    faceid=YES|NO|unset
    health=YES|NO|unset (iOS/tvOS 12.0 and above)
    homekit=YES|NO|unset
    location=always|inuse|never|unset
    notifications=YES|NO|critical|unset
    speech=YES|NO|unset
    userTracking=YES|NO|unset (iOS/tvOS 14.0 and above)
```

Reply:
I believe we need to test Apple's privacy thing for all possible permission types. Also, as far as I remember, simctl forcefully restarts app under test. Not sure wix's utility requires that.

Reply:
Actually Apple's one reset while the wix did not. Not sure if these process restart affects permission's behavior, but potentially we could use only `location` with the privacy method

Reply:
Except for `contacts`, Apple's one worked but it reset the app process. So... probably we can use the apple's one only for `locations`

Reply:
Updated. So, the latest implementation uses the privacy command only for the location stuff. Other commands by privacy works except for contacts, but the privacy command kills the app process, which might be breaking changes in terms of switching the method from wix.